processing/src/main/java/org/apache/druid/query/IterableRowsCursorHelper.java

```diff
@@ -55,8 +64,10 @@ public static RowBasedCursor<Object[]> getCursorFromIterable(Iterable<Object[]>
       }
       return row -> row[columnIndex];
     };
-    RowWalker<Object[]> rowWalker = new RowWalker<>(Sequences.simple(rows), rowAdapter);
-    return new RowBasedCursor<>(
+
+    RowWalker<Object[]> rowWalker = new RowWalker<>(rows, rowAdapter);
```

Comment:
Since you need to close the RowWalker, why don't you change `RowBasedCursor` and add a closeable to it. 

In its close method, call rowWalker.close().

Reply:
Revisiting this comment, it doesn't seem fair that the `RowBasedCursor` should close the `rowWalker` that was passed to it by the caller. The caller should close the `rowWalker`.

Either the method `getCursorFromSequence` should return a Pair<Cursor, Closeable> or I will rename `CloseableCursor` to `CursorWithCloseable`